### DISCUSSION TITLE
Generate structs with x/tools fieldalignment suggested ordering

### DISCUSSION THREAD

**Discussion by natethinks** [sqlc-dev/sqlc#1281] [Feature Requests & Ideas]
x/tools `fieldalignment -fix` will reorder your structs to decrease memory usage and garbage collection. I propose the `fieldallignment` order be used in the structs generated by sqlc. This can be run manually with no side effects as far as I can tell, but you'll have to run it again every time you generate the code.

**Comment by kevinburke** [comment]
I think you'd want to trade this off with the readability of having e.g. the important fields in a database model listed first, or consecutively. In practice maybe it doesn't matter because the fields are not moving around _that_ much.

**Reply by natethinks** [reply]
That's fair. Although I'm not sure how much I value readability in generated code. I don't really care what the the generated structs look like as long as they accurately represent my DB table structure.

**Comment by ksthiele** [comment]
+1  it could save a good amount of memory, even if its just optional

https://medium.com/@felipedutratine/how-to-organize-the-go-struct-in-order-to-save-memory-c78afcf59ec2
